### PR TITLE
Enhance min frequency checking

### PR DESCRIPTION
**Pull request by takojunior**
Highlighted updates:
- Raise a value error when min_frequency is a float, while the minimum row count is set to be less than 1.
- Make the item-integer_ID map to be deterministic and the order of output patterns get fixed when sequences use characters.
- Add unit tests for the corner cases, such as empty list, one or two rows with min_frequency being a percentage.
- Update version to 1.2.1 and the docs.

### CODE REVIEW

File: CHANGELOG.txt
Comment (skadio) [inline CHANGELOG.txt:9]: This is probably not a "major" issue. It is basically an input validation.

File: sequential/utils.py
Comment (skadio) [inline sequential/utils.py:135]: just checking my understanding: why sort (set) is not enough? And we use sort(list(set)) instead?

**Review by skadio — APPROVED** [review]
Looks good to me

**Comment by skadio** [comment]
One question: why the change on HTML files?

File: sequential/utils.py
Comment (takojunior) [reply sequential/utils.py:135]: Yep, you are right. There is no need to use list(). Sort(set) will work and give the same results.

**Comment by takojunior** [comment]
The docs also contains a page for the source code of sequential.seq2pat, then we might want to update the html file if there is an update on the code. So I recreate the docs.

File: CHANGELOG.txt
Comment (takojunior) [reply CHANGELOG.txt:9]: Make sense, will change these to minor.

**Comment by takojunior** [comment]
@skadio, minor updates have been made at the places regarding the comments, see this [commit](https://github.com/fidelity/seq2pat/pull/15/commits/50ec174de9e3395d7d15ba6badb02b44dfd612cb).

File: sequential/seq2pat.py
Comment (skadio) [inline sequential/seq2pat.py:386]: don't we need a check_true (self._num_rows >= 1) independent of if (float) else (int) logic?

**Review by skadio — APPROVED** [review]
Approved from my end. Feel free to merge as you proceed, no need for another review round. 

Does this require recompilation of the backend etc? I don't remember that part.

I assume you will update the PyPI package part once this is merged. And we can follow-up to that issue thread to announce the fix and Close the open Issue.

**Comment by takojunior** [comment]
> Approved from my end. Feel free to merge as you proceed, no need for another review round.
> 
> Does this require recompilation of the backend etc? I don't remember that part.
> 
> I assume you will update the PyPI package part once this is merged. And we can follow-up to that issue thread to announce the fix and Close the open Issue.

This doesn't require recompilation since backend hasn't been changed. The Python part takes care of the input validation. And after the merge, I can update PyPI release and follow up the issue.

File: CHANGELOG.txt
Comment (takojunior) [reply CHANGELOG.txt:11]: > you mean less than or equal?

Equal could be ok since it will allow the minimum threshold of row count to be 1. So when it is an equal, it is not throwing an error.

File: sequential/seq2pat.py
Comment (takojunior) [reply sequential/seq2pat.py:386]: > don't we need a check_true (self._num_rows >= 1) independent of if (float) else (int) logic?

I think this might be covered already. We checked the input needs to be a list of list, and input has no empty list. But this can be added to make it explicit.

Comment (takojunior) [reply sequential/seq2pat.py:388]: Added a check when sequences/attributes are initialized with an empty list, and also have `num_rows` checked.

**Pull request merged**
